beacon_node/beacon_chain/src/builder.rs

```diff
@@ -325,9 +325,20 @@ where
                 .unwrap_or_else(OperationPool::new),
         );
 
-        let pubkey_cache = ValidatorPubkeyCache::load_from_store(store)
+        let mut pubkey_cache = ValidatorPubkeyCache::load_from_store(store.clone())
             .map_err(|e| format!("Unable to open persisted pubkey cache: {:?}", e))?;
 
+        // If any validators weren't persisted to disk on previous runs, this will use the genesis_state to
+        // "top-up" the in-memory validator cache and its on-disk representation with any missing validators.
+        let pubkey_store_ops = pubkey_cache
+            .import_new_pubkeys(&genesis_state)
```

Comment:
I don't think we wanna do this with the genesis state, we need the head state (or maybe the finalized state, but the head would be better)

Reply:
Here: https://github.com/sigp/lighthouse/blob/491091f317371aea92110487b6e8a0c960744a49/beacon_node/beacon_chain/src/builder.rs#L817-L820

Reply:
We'd do the top up in the `Ok` case here: https://github.com/sigp/lighthouse/blob/491091f317371aea92110487b6e8a0c960744a49/beacon_node/beacon_chain/src/builder.rs#L861-L864